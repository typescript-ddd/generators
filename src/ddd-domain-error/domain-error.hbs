import { DomainError } from "@typescript-ddd/core";

/**
 * Represents {{noun errorName}} error.
 */
export class {{pascal errorName}}Error extends DomainError {
  /**
   * Initializes a new instance of {{noun errorName}} error.
   {{#ifUndefined message}}* @param { string | undefined } message - The message text.{{newline}}*/{{else}}*/{{/ifUndefined}}
  {{#ifDefined message}}
  public constructor() {
    super("{{../message}}");
  }
  {{else}}
  public constructor(message?: string) {
    super(message ?? "{{firstUpper (noun ../errorName)}} error has occurred.");
  }
  {{/ifDefined}}
}