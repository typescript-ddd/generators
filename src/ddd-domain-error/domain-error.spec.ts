import { DomainErrorTemplateData } from "./domain-error.types";
import { generateDomainError } from "./domain-error.generator";

describe("EntityId", () => {
  it("should render domain error template with message", () => {
    // Arrange
    const data: DomainErrorTemplateData = {
      errorName: "Unauthorized",
      message: "User is not authorized to perform this action",
    };

    const result = generateDomainError(data);
    expect(result).toMatchSnapshot();
  });

  it("should render domain error template with message parameter", () => {
    // Arrange
    const data: DomainErrorTemplateData = {
      errorName: "Unauthorized",
    };

    const result = generateDomainError(data);
    expect(result).toMatchSnapshot();
  });

  it("should remove error suffix if present", () => {
    // Arrange
    const data: DomainErrorTemplateData = {
      errorName: "UnauthorizedError",
    };

    const result = generateDomainError(data);
    expect(result).toMatchSnapshot();
  });
});
