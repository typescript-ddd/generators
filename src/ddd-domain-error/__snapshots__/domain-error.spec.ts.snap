// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityId should remove error suffix if present 1`] = `
"import { DomainError } from "@typescript-ddd/core";

/**
 * Represents a unauthorized error.
 */
export class UnauthorizedError extends DomainError {
  /**
   * Initializes a new instance of a unauthorized error.
   * @param { string | undefined } message - The message text.
*/
  public constructor(message?: string) {
    super(message ?? "A unauthorized error has occurred.");
  }
}"
`;

exports[`EntityId should render domain error template with message 1`] = `
"import { DomainError } from "@typescript-ddd/core";

/**
 * Represents a unauthorized error.
 */
export class UnauthorizedError extends DomainError {
  /**
   * Initializes a new instance of a unauthorized error.
   */
  public constructor() {
    super("User is not authorized to perform this action");
  }
}"
`;

exports[`EntityId should render domain error template with message parameter 1`] = `
"import { DomainError } from "@typescript-ddd/core";

/**
 * Represents a unauthorized error.
 */
export class UnauthorizedError extends DomainError {
  /**
   * Initializes a new instance of a unauthorized error.
   * @param { string | undefined } message - The message text.
*/
  public constructor(message?: string) {
    super(message ?? "A unauthorized error has occurred.");
  }
}"
`;
